HEN_HOUSE/egs++/ausgab_objects/egs_dose_scoring/egs_dose_scoring.cpp

```diff
@@ -469,20 +469,41 @@ void EGS_DoseScoring::outputDoseFile(const EGS_Float &normD) {
 }
 
 bool EGS_DoseScoring::storeState(ostream &data) const {
+    // Convert the ostream to FILE * before writing data
+    // This might seem like a hack, but improves efficiency for writing
+    // very large arrays (like 3ddose) to egsdat
+    // It's otherwise fine (or better) to use ostream elsewhere
+
+    // Get the position from the stream
+    long pos = data.tellp();
+    data.flush();
+
+    string ofile = app->constructIOFileName(".egsdat",true);
```

Comment:
While I appreciate the gain in performance in writing this via a separate `FILE *`,  it makes the code fragile here. There is no guarantee that the hard-coded filename (which cannot be recovered from the stream) will match that of the stream, and the code relies on perfect synchronization between the two separate file handles, which is not guaranteed and  even OS or compiler dependent.

I reckon the only place where the ostream proves really slow for large arrays is the repeated calls inside the scoreSingle method. I think we can improve it _there_, by buffering (or double-buffering) the write loop in the scoring array storeState method and realtively few `ostream::write` calls.

I propose to merge in only the innocuous `<< endl;` changes at this time, and keep the proposed `FILE *` on this branch for further testing along the lines above.